File: benchmarks/loop_sum.go
package main

import (
	"fmt"
	"time"
)

func main() {
	t0 := time.Now()
	s := 0
	i := 0
	for i < 100000000 {
		s = s + i
		i = i + 1
	}
	fmt.Println(s)
	fmt.Printf("bench_ns=%d\n", time.Since(t0).Nanoseconds())
}

File: benchmarks/loop_sum.rs
fn main() {
    let t0 = std::time::Instant::now();
    let mut s: i64 = 0;
    let mut i: i64 = 0;
    while i < 100000000 {
        s = s.wrapping_add(i);
        i += 1;
    }
    let duration = t0.elapsed();
    println!("{}", s);
    println!("bench_ns={}", duration.as_nanos());
}
